### PR TITLE
Merge PersistentActiveAccountHolder into CodyAuthenticationManager

### PR DESCRIPTION
**Pull request by mkondratek**
Fixes https://linear.app/sourcegraph/issue/CODY-2666/merge-codyprojectactiveaccountholder-and-codyauthenticationmanager.

## Test plan

Authentication 
- log in for the first time 
- having multiple accounts
- logging into the same account twice
- switching accounts b/w pro, free, enterprise

### CODE REVIEW

File: src/main/kotlin/com/sourcegraph/cody/auth/SelectOneOfTheAccountsAsActive.kt [hunk 1]
```diff
@@ -3,24 +3,19 @@ package com.sourcegraph.cody.auth
 import com.intellij.openapi.project.Project
 import com.sourcegraph.cody.CodyToolWindowContent
 import com.sourcegraph.cody.config.CodyAuthenticationManager
-import com.sourcegraph.cody.config.CodyProjectActiveAccountHolder
 import com.sourcegraph.cody.config.getFirstAccountOrNull
 import com.sourcegraph.cody.initialization.Activity
 
 class SelectOneOfTheAccountsAsActive : Activity {
```
Comment (pkukielka) [inline src/main/kotlin/com/sourcegraph/cody/auth/SelectOneOfTheAccountsAsActive.kt]: Cannot we now move that whole code to `CodyAuthenticationManager::loadState`?

Comment (mkondratek) [reply src/main/kotlin/com/sourcegraph/cody/auth/SelectOneOfTheAccountsAsActive.kt]: good point, lemme test it...

**Review by pkukielka — APPROVED** [review]
LGTM, thanks!
Also `+84 −155` ❤️

File: src/main/kotlin/com/sourcegraph/cody/auth/SelectOneOfTheAccountsAsActive.kt (hunk 1 above)
Comment (mkondratek) [reply src/main/kotlin/com/sourcegraph/cody/auth/SelectOneOfTheAccountsAsActive.kt]: needed to override noStateLoaded to get it working without the active account set initially (loadState executes only if there is a state)

docs:
https://plugins.jetbrains.com/docs/intellij/persisting-state-of-components.html#customizing-the-xml-format-of-persisted-values

**Pull request merged**